Sources/SourceKitLSP/TestDiscovery.swift

```diff
@@ -149,7 +149,18 @@ extension SourceKitLSPServer {
           utf16index: testSymbolOccurrence.location.utf8Column - 1
         )
       }
-      let id = (context + [testSymbolOccurrence.symbol.name]).joined(separator: "/")
+
+      var symbolContext = context
+
+      // If the symbol is defined in an extension we need to augment the ID with the context.
+      let parentExtension = testSymbolOccurrence.relations.first(where: {
+        $0.symbol.kind == IndexSymbolKind.extension
+      })
+      if let parentExtension {
+        symbolContext.append(parentExtension.symbol.name)
+      }
```

Comment:
I think it would be better to retrieve the type extended by the extension when we build up the context then doing something special here.

Reply:
This does produce the correct id, but the AnnotatedTestItem returned on line 179 still has `isExtension: false` which IIRC will affect the sort order of the tests.

Reply:
Good, point. Wasn’t really relevant in this test but it did cause us to not report tests from extensions if the definition of the test was in a different file. Updated the PR to take care of that.